LoopKitUI/Views/Settings Editors/SuspendThresholdEditor.swift

```diff
@@ -12,59 +12,51 @@ import LoopKit
 
 
 public struct SuspendThresholdEditor: View {
-    var initialValue: HKQuantity?
-    var unit: HKUnit
-    var maxValue: HKQuantity?
-    var save: (_ suspendThreshold: HKQuantity) -> Void
-    let mode: SettingsPresentationMode
-    
+    @Environment(\.dismiss) var dismiss
+    @Environment(\.authenticate) var authenticate
+    @Environment(\.appName) private var appName
+
+    @ObservedObject var viewModel: TherapySettingsViewModel
```

Comment:
I'd avoided using the `viewModel` here in case some day in the future this Editor might be reused outside of Therapy Settings.  That future day has yet to arrive, and this is not a blocking comment, I just wanted to give you some context and history.

Reply:
I'm didn't comment on this earlier, but I think the general usage of the `TherapySettingsViewModel` for all of the different views is too broad. To me, each view should have its own view model so the two an be used independently of the therapy settings acceptance workflow. Just my two cents...

Reply:
Agreed.  But not all Views have ViewModels... some were created with a few pieces of `@State` and that's it (as was the case for this Editor).  The `init` that takes a `TherapySettingsViewModel` was created as a convenience `init`...and probably shouldn't have been put in this file.

Reply:
In my opinion, the view was already relying on the view model (save, establish initial values, etc). The only reason this view exists is to edit a value provided by the view model and have the view model save it. So it made sense that it retains the view model. As an added benefit, it is able to observer and respond to changes in the view model. 

Happy to discuss this after standup.

Reply:
I agree with the above comments that having these models rely on the full TherapySettingsViewModel is not great. That's a much bigger set of dependencies than what this view really needs.